TopMovies/View/CategoryVC/MoviesCategoryVC.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  MoviesCategoryVC.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 17.02.2021.
+//
+
+import UIKit
+import SnapKit
+
+// MARK: - Props struct
+struct MoviesCategoryVCProps {
+  let categoryId: MovieCategory.ID
+  let categoryName: String
+  let loadProgress: LoadProgress
+  var moviesState: MoviesState
+  
+  struct LoadProgress {
+    let isReloadInProgress: Bool
+    let isLoadMoreInProgress: Bool
+  }
+  struct MoviesState {
+    let hasNewData: Bool
+    let moviesUsage: RequestType
+    var movies: [MovieTableViewCellProps]
+  }
+}
+
+// MARK: - VC class
+class MoviesCategoryVC: UIViewController, PropsConnectable {
+  var propsConnector: BaseConnector<MoviesCategoryVCProps>?
+  var props = MoviesCategoryVCProps(categoryId: MovieCategory.ID(value: ""),
+                                    categoryName: "",
+                                    loadProgress: .init(isReloadInProgress: false,
+                                                        isLoadMoreInProgress: false),
+                                    moviesState: .init(hasNewData: false,
+                                                       moviesUsage: .reload,
+                                                       movies: [])) {
+    didSet {
+      title = props.categoryName
+      categoryTableView.reloadData()
+    }
+  }
+  private let categoryTableView = UITableView(frame: .zero, style: .grouped)
+  private let refreshControl = UIRefreshControl()
+  private let footerIndicatorView = UIActivityIndicatorView()
+  private let movieCellIdentifier = String(describing: MovieTableViewCell.self)
+  private let cellHeight: CGFloat = 200.0
+  
+  // MARK: - Setup Connection
+  public func configureConnectionWith(connector: BaseConnector<MoviesCategoryVCProps>) {
+    propsConnector = connector
+  }
+  public func connect(props: MoviesCategoryVCProps) {
+    handleRefresh(props.loadProgress)
+    if props.moviesState.hasNewData {
+      switch props.moviesState.moviesUsage {
+      case .reload:
+        self.props = props
+      case .loadMore:
+        self.props.moviesState.movies += props.moviesState.movies
```

Comment:
That isn't where you have to decide what movies to display.

UI should be fully represented by props. That's the difference between state and props. The state contains all data and the connector decides how it should be shown on UI.

Eventually, you get `props` that should only be set to VC without any additional logic. All calculations should be done in the connector.